qsoverlay/circuit_builder.py

```diff
@@ -14,45 +14,17 @@
 
 class Builder:
 
-    def __init__(self,
-                 setup=None,
-                 qubit_dic=None,
-                 gate_dic=None,
-                 gate_set=None,
-                 update_rules=None,
-                 **kwargs):
+    def __init__(self, setup, **kwargs):
         '''
-        qubit_dic: list of the qubits in the system.
-            Each qubit should have a set of parameters,
-            which is called whenever required by the gates.
-        gate_dic: a dictionary of allowed gates.
-            Each 'allowed gate' consists of:
-                - the function to be called
-                - a set of 'qubit_args' that will be found in
-                    the qubit dictionary and passed to the gate.
-                - a 'time' - the length of the gate
-            Note that 'Measure' should be in the gate_set
-        gate_set: a dictionary of allowed gate instances.
-            An allowed gate instance is a list of the gate
-            along with the qubits it is performed between.
-        update_rules: a set of rules for updating the system.
-            (i.e. between experiments).
+        Args:
+        -------
+        setup: an experiment_setup.Setup containing details of
+            the experiment.
 
         kwargs: Can add t1 and t2 via the kwargs instead of
             passing them with the qubit_dic.
```

Comment:
Kinda weak explanation here (I would not be able to understand). May be better link it to Quantumsim circuit documentation?